New outfit tutorial

JojjeE opened the pull request:
Covers:
- Charset
- Outfit
- Globaldata
- Blobs
- Localization
- Peacock (Repository and unlockables)
- Chongqing rain logic

I think this is our longest yet clocking in at a little under 6800 words. Please give feedback if anything looks weird or I haven't explained something well enough. Brain mushy. Thanks.

JojjeE committed:
begin draft on new outfit tutorial

JojjeE committed:
bump framework version

JojjeE committed:
finish preplanning for newoutfit.md

JojjeE committed:
Charset part

JojjeE committed:
Begin constructing outfit

JojjeE committed:
Outfit and globaldata stuff

JojjeE committed:
Replacing inventory image and localization strings

JojjeE committed:
extra credit: Peacock addon

JojjeE committed:
Chongqing rain effects

JojjeE committed:
Apparently patching unlockables starter package is completely unnecessary

Thanks to IKennyAgain#1337 for this revelation and I feel like a goof for not realizing it sooner

Notexe commented on docs/modding/suitmodding/newoutfit.md:
```diff
@@ -0,0 +1,648 @@
+
+## Extra Credit: Peacock Addon
+
+Your mod is finished now, if your goal was just to replace a suit with your own. But we can go the extra mile too, and add our suit to the game without replacing any others. This will only work if the mod user is playing on Peacock, however. Doing this will require modifying the repository and unlockables which won't do anything if we play on IOI's servers, as we mentioned earlier.
```
Might be worth mentioning that this also works in offline mode

**Edit:** nvm that requires the starter package thing I think

JojjeE replied:
Ah I didn't consider offline mode. Yeah that would be the purpose of the starter package patch. I already wrote it so I may as well reincorporate it slightly reworded.

Notexe reviewed (commented):
From what I can tell everything seems to be good. I'm not really too sure if it's worth adding information about patching the starter package thing for offline mode support or not.

JojjeE committed:
Reincorporate starter package patching for offline

Also alter some wording and a tip about checking globaldata if you can't find an outfit in the spreadsheet.

Notexe commented:
Seems good to me, will merge it
